Mage.Sets/src/mage/cards/p/PoolOfVigorousGrowth.java

```diff
@@ -0,0 +1,115 @@
+package mage.cards.p;
+
+import java.util.List;
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.common.ActivateAsSorceryActivatedAbility;
+import mage.abilities.costs.common.DiscardCardCost;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.*;
+import mage.cards.repository.CardCriteria;
+import mage.cards.repository.CardInfo;
+import mage.cards.repository.CardRepository;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.game.Game;
+import mage.game.permanent.token.Token;
+import mage.game.permanent.token.custom.CreatureToken;
+import mage.players.Player;
+import mage.util.RandomUtil;
+import mage.util.functions.CopyTokenFunction;
+
+/**
+ *
+ * @author karapuzz14
+ */
+public final class PoolOfVigorousGrowth extends CardImpl {
+
+    public PoolOfVigorousGrowth(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{1}{G}");
+        
+
+        // {X}, {T}, Discard a card: Create a token that's a copy of a random creature card with mana value X. Activate only as a sorcery.
+        Ability ability = new ActivateAsSorceryActivatedAbility(
+                new PoolOfVigorousGrowthEffect(), new ManaCostsImpl<>("{X}")
+        );
+        ability.addCost(new TapSourceCost());
+        ability.addCost(new DiscardCardCost());
+        this.addAbility(ability);
+    }
+
+    private PoolOfVigorousGrowth(final PoolOfVigorousGrowth card) {
+        super(card);
+    }
+
+    @Override
+    public PoolOfVigorousGrowth copy() {
+        return new PoolOfVigorousGrowth(this);
+    }
+}
+class PoolOfVigorousGrowthEffect extends OneShotEffect {
+
+    PoolOfVigorousGrowthEffect() {
+        super(Outcome.PutCardInPlay);
+        staticText = "Create a token that's a copy of a random creature card with mana value X";
+    }
+
+    private PoolOfVigorousGrowthEffect(final PoolOfVigorousGrowthEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public PoolOfVigorousGrowthEffect copy() {
+        return new PoolOfVigorousGrowthEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+        
+        int xValue = source.getManaCostsToPay().getX();
+        if (game.isSimulation()) {
+            // Create dummy token to prevent multiple DB find cards what causes H2 java.lang.IllegalStateException if AI cancels calculation because of time out
```

Comment:
Can you explain details about broken AI here? Is it catch error all time or at random? How to reproduce that?

Reply:
Oh, that's just old comment from Momir's Emblem implementation. I decided that it's known issue and just left the same comment as explanation.
I think it says about next issue:
https://github.com/magefree/mage/issues/7014

Reply:
So this trick with dummy token is also stolen from here:
https://github.com/magefree/mage/blob/da48821754dbf37bc1382775f9e0a125d8e2cfe3/Mage/src/main/java/mage/game/command/emblems/MomirEmblem.java#L76